tests/ServiceTokenVoter/AbstractReadonlyBase.php

```diff
@@ -17,22 +17,7 @@ public function setUp(): void
         $this->voter = new Voter();
     }
 
-    public function supportsTypeProvider(): array
```

Comment:
Am I understanding correctly that this went away because it's only used in `GreenLightViewDTOVoter` and `ReadonOnlyDTOVoter` so isn't inherited elsewhere?

Reply:
this had to be re-tooled, unfortunately. since the previous method was invoking an abstract method (another data provider), which isn't allowed with  _abstract_ static methods.
so i had to push this down into the concrete test classes.

Reply:
Ah, yeah that tracks. 🆒